### PR TITLE
XP-4889 Option Set - Don't clear input values when an option is unselected

### PR DESCRIPTION
**Pull request by sshklianko**

### CODE REVIEW

File: modules/admin/admin-ui/src/main/resources/web/admin/common/js/form/set/optionset/FormOptionSetOptionView.ts [hunk 1]
```diff
@@ -281,9 +295,45 @@ module api.form {
...
+
+        private isChildOfDeselectedParent(): boolean {
+            var result = false;
+
+            wemjq(this.getEl().getHTMLElement()).parents('.form-option-set-option-view').each((index, elem) => {
+                if(!wemjq(elem).hasClass('selected')) {
+                    result = true;
+                }
+            });
+
```
Comment (alansemenov) [inline modules/admin/admin-ui/src/main/resources/web/admin/common/js/form/set/optionset/FormOptionSetOptionView.ts]: This looks complicated. You could just do
```
return !wemjq(this.getEl().getHTMLElement()).parents('.form-option-set-option-view').hasClass('selected');
```

File: modules/admin/admin-ui/src/main/resources/web/admin/common/js/form/set/optionset/FormOptionSetOptionView.ts [hunk 2]
```diff
@@ -381,8 +437,11 @@ module api.form {
         private updateViewState() {
             this.expand(this.isOptionSetExpandedByDefault() || this.getThisPropertyFromSelectedOptionsArray() != null);
 
-            if (this.isOptionSetExpandedByDefault() && this.getThisPropertyFromSelectedOptionsArray() == null) {
-                this.disableFormItems();
+            if (this.getThisPropertyFromSelectedOptionsArray() == null) {
+                if (this.isOptionSetExpandedByDefault()) {
+                    this.disableFormItems();
+                }
+                this.cleanValidationForThisOption();
```
Comment (alansemenov) [inline modules/admin/admin-ui/src/main/resources/web/admin/common/js/form/set/optionset/FormOptionSetOptionView.ts]: Can we replace double calls to isOptionSetExpandedByDefault() and getThisPropertyFromSelectedOptionsArray() with local variables? Also replace comparison with null with (!)

File: modules/admin/admin-ui/src/main/resources/web/admin/common/js/form/set/optionset/FormOptionSetOptionView.ts [hunk 3]
```diff
@@ -299,6 +349,13 @@ module api.form {
             this.removeClass('invalid');
         }
 
+        private cleanSelectionMessageForThisOption() {
+
+            wemjq(this.getEl().getHTMLElement()).find('.selection-message').each((index, elem) => {
+                wemjq(elem).addClass('empty');
+            });
+        }
+
```
Comment (alansemenov) [inline modules/admin/admin-ui/src/main/resources/web/admin/common/js/form/set/optionset/FormOptionSetOptionView.ts]: Same thing here. Just do
```
wemjq(this.getEl().getHTMLElement()).find('.selection-message').addClass('empty');
```

**Comment by sshklianko** [comment]
Made a new PR (https://github.com/enonic/xp/pull/4357) with adjustments you proposed except moving getThisPropertyFromSelectedOptionsArray() to a local variable - this value is taken from parentDataSet object that may be replaced (during update() for example)

**Pull request closed without merging**